### PR TITLE
NO-TICKET: Limit message size in chainspec

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This ensures that nodes are in agreement about what the maximum message size is through the chainspec.

### CODE REVIEW

**Comment by goral09** [comment]
I don't think this should be a chainspec config after all. I think the networking should allow arbitrarily large messages and if there's a business/domain limit then that should be applied in a proper component rather than the networking layer.

**Comment by marc-casperlabs** [comment]
> I don't think this should be a chainspec config after all.

This is somewhat debateable, but the maximum size of anything sent across the network depends on values from the chainspec, so in theory we should be able to calculate it from the chainspec anyway. Right now, we just shortcut the otherwise somewhat complex estimate by putting in an explicit value.

It would be perfectly fine to have this value configurable by node operators, except that it would not gain us anything while at the same potentially breaking things. If nodes differ in their opinion on the maximum acceptable message size for network messages, we will run into issues.

To the best of my knowledge, the chainspec contains "values all nodes need to agree on" as a rule of thumb, so using that definition, I think it is a good place to keep it.

> I think the networking should allow arbitrarily large messages 

We can never allow arbitrarily large messages because we cannot afford to spend arbitrarily large amounts of memory before starting to validate contents. If large sizes (the current limit is an initial one, bound to come down soon) are required for contents, a chunking scheme will have to be implemented (which is in the works, just not for this release).

> and if there's a business/domain limit then that should be applied in a proper component rather than the networking layer.

This is purely a layer in the defense against a resource exhaustion; the acceptable size for a deploy or other things is still not determined by the networking layer. There just has to be _some_ bound for the network to function, otherwise nodes are trivially crashable.

File: node/src/components/small_network.rs [hunk 1]
```diff
@@ -550,6 +556,7 @@ where
         effect_builder: EffectBuilder<REv>,
         peer_id: NodeId,
         transport: Transport,
+        maximum_net_message_size: u32,
```
Comment (goral09) [inline node/src/components/small_network.rs:559]: Does it have to be an argument? Can't you take that from `self`?

**Review by goral09 — APPROVED** [review]

File: node/src/components/small_network.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components/small_network.rs:559]: Good catch!

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**